### PR TITLE
[MRG] Update cli docs

### PR DESCRIPTION
**Pull request by ctb**
A first attempt at #1092.

Also adds a table of contents to the command-line docs, which is pretty nice!

- [x] Is it mergeable?
- [x] `make test` Did it pass the tests?
- [x] `make coverage` Is the new code covered?
- [x] Did it change the command-line interface? Only additions are allowed
  without a major version increment. Changing file formats also requires a
  major version number increment.
- [x] Was a spellchecker run on the source code and documentation after
  changes were made?

### CODE REVIEW

**Comment by ctb** [comment]
Ready for review @bluegenes. happy to iterate a few times and add more info :)

**Comment by ctb** [comment]
(although there is a question as to whether we should merge these docs just yet, since right now the docs build from master, not from a stable branch that tracks the latest release.)

File: doc/command-line.md [hunk 1]
```diff
@@ -758,3 +782,113 @@ For example,
...
+* `compare` takes multiple signatures and can load them from files,
+  directories, and indexed databases (SBT or LCA).  It can also take
+  a list of file paths in a text file, using `--from-file` (see below).
+  
+* the `lca classify` and `lca summarize` commands take multiple
+  signatures with `--query`, and multiple LCA databases, with
+  `--db`. `sourmash multigather` also uses this style.  This allows these
+  commands to specify multiple queries **and** multiple databases without
+  (too much) confusion.  These commands will take files containing
+  signature files using `--query-from-file` (see below).
```
Comment (bluegenes) [inline doc/command-line.md:810]: do these search multiple ksizes /scaled?

File: doc/command-line.md [hunk 2]
```diff
@@ -758,3 +782,113 @@ For example,
...
+(LCA databases also permit taxonomic searches using `sourmash lca` functions.)
+
+The main point is that since all of these databases contain signatures,
+as of sourmash 3.4, any command that takes more than one signature will
+also automatically load all of the signatures in the database.
+
+Note that, for now, both SBT and LCA database can only contain one
+"type" of signature (one ksize, one moltype, etc.) If the database
+signature type is incompatible with the other signatures, sourmash
+will complain.  will be raised.  In contrast, signature files can
```
Comment (bluegenes) [inline doc/command-line.md:843]:
```suggestion
will complain. In contrast, signature files can
```

Unless you wanted to mention the specific error?

File: doc/command-line.md (hunk 1 above)
Comment (ctb) [reply doc/command-line.md:810]: no, they don't - will update.

File: doc/command-line.md (hunk 2 above)
Comment (ctb) [reply doc/command-line.md]: The error message should be self-explanatory, but I'll check.

Comment (ctb) [reply doc/command-line.md]: whoops, nice catch, found a bug 😂

**Pull request merged**